R/metric-selection.R

```diff
@@ -58,6 +67,47 @@ contains_survival_metric <- function(mtr_info) {
   any(grepl("_survival", mtr_info$class))
 }
 
+#' @rdname choose_metric
+#' @export
+choose_eval_time <- function(x, metric, eval_time = NULL, call = rlang::caller_env()) {
+  mtr_set <- .get_tune_metrics(x)
+  mtr_info <- tibble::as_tibble(mtr_set)
+
+  if (!contains_survival_metric(mtr_info)) {
+    return(NULL)
+  }
+
+  # If we need an eval time, set it to the possible values so that
+  # we can choose the first value
+  if (is_dyn(mtr_set, metric) & is.null(eval_time)) {
```

Comment:
```suggestion
  if (is_dyn(mtr_set, metric) && is.null(eval_time)) {
```

Reply:
Why not `&&`?

Reply:
I'm suggesting `&&` because it enforces a single TRUE/FALSE so that it errors on the comparison rather than the `if` statement it's in

``` r
rep(TRUE, 2) & rep(FALSE, 2)
#> [1] FALSE FALSE
rep(TRUE, 2) && rep(FALSE, 2)
#> Error in rep(TRUE, 2) && rep(FALSE, 2): 'length = 2' in coercion to 'logical(1)'
```

<sup>Created on 2023-12-06 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>